src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreResult.cs

```diff
@@ -230,6 +219,18 @@ public virtual async Task CommitAsync(ILogger log, CancellationToken token)
                         result.LockFilePath));
                 }
             }
+
+            // write packages lock file if enabled.
+            if (NewPackagesLockFile != null && !string.IsNullOrEmpty(NewPackagesLockFilePath))
```

Comment:
should this be outside of this method, since the method name is `CommitAssetsFileAsync` (or change the name of the method) and the whole reason of the PR is to make assets file and lock file independent?

Reply:
sounds good, I'll create a separate method for this.